packages/graph/hash_graph/lib/graph/src/shared/subgraph/edges/edge.rs

```diff
@@ -63,18 +63,6 @@ impl ToSchema for OntologyOutwardEdges {
 #[derive(Debug, Hash, PartialEq, Eq, Serialize)]
 #[serde(untagged)]
 pub enum KnowledgeGraphOutwardEdges {
-    ToKnowledgeGraph(OutwardEdge<KnowledgeGraphEdgeKind, EntityIdAndTimestamp>),
+    ToKnowledgeGraph(OutwardEdge<KnowledgeGraphEdgeKind, EntityId>),
     ToOntology(OutwardEdge<SharedEdgeKind, OntologyTypeEditionId>),
 }
```

Comment:
I appreciate the optimisation we're getting from this change but the separation between the representations is increasing more and more and I really don't like what we're doing for maintainability of an already very complicated object.

(I'm not suggesting we shouldn't do this by the way, just I want us to be aware of it)

Reply:
I think it makes sense to use the `crate::subgraph::Subgraph` struct as the structure to build the subgraph, not as the subgraph itself. Maybe we define that in the `DependenyContext` (or `TraversalContext` in the future) and allow creating the `Subgraph` as defined in the `uitoipa_typedefs` module as the "real" subgraph. Then we would have the same subgraph available everywhere. 
If you are happy with this approach I will do that in a follow-up.

Reply:
I think that would make sense except for what I spoke about before:
> Furthermore, if we actually do start interacting with the subgraph in Rust (which I am pretty confident we will), then we do not have the appropriate information right now to actually use it. (This is also why I disliked us getting rid of Deserialize because it's easy for us to make the mapping only work one-way easily).

I don't think we'd want to interface with the utoipa-polluted representation, I think we'd want to use the cleaner more native-friendly representation.

That's why I think we should go for this:
> I think if we move to BTreeMaps then we can have the earliest thing be done as part of querying the native object (e.g. getEdgesForElement or something), rather than a post-processing step, as it should be quick enough. We could then maybe use that same logic in an `Into` implementation, which could be consistent across representations (and used in the JSON representation), rather than the way we're doing it now which would require us to reimplement it if we serialized to flatbuffers for example.

So, instead of calling `earliest_entity_by_id` in the `Serialize` implementation, I think we should do it as part of an `IntoIterator` for the native `Edges` or something and `FromIterator`. Then we'd benefit from it on both sides.